examples/ffi/python_calling_rust/py_library_from_rust.bzl

```diff
@@ -0,0 +1,61 @@
+# TODO: Native extensions are what we really want, cdll.LoadLibrary is not particularly great.
+def py_library_from_rust(name, rust_lib):
+    # 1. Put the library into _solib_ because of the ODR.
+    #
+    # nb. Without this, the .so doesn't make it to runfiles at all
+    native.cc_library(
+        name = "c_wrapper",
+        srcs = [rust_lib],
+    )
+
+    # 2. Elaborate dance to figure out the path to the shared object library in _solib_...
+    crate_name = rust_lib.split(":")[-1]
+    out = name + ".py"
+    native.genrule(
+        name = "__generate" + out,
+        srcs = [rust_lib],
+        outs = [out],
+        # TODO: Was there a template rule somewhere?
+        cmd = """
+        cat << EOM > $(OUTS)
+
+from __future__ import print_function
+
+import os
+from ctypes import cdll
+
+# TODO: This is a hack to compensate for rootpath not giving us the _solib_* path...
```

Comment:
I cannot quite put my finger on how hacky this is.. let me know what ya'll think.

Reply:
This ultimately searches all of runfiles, so unless the design of collecting shared libraries in one location changes dramatically, it should be safe. That said, searching should really not be necessary and I'm not sure if it's a deficiency in python/c++ rules or my lack of knowledge here.